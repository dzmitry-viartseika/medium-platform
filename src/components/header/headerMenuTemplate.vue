<template>
  <div class="app-menu">
    <template v-if="typeof userInfo === 'object'">
      <template v-if="!Object.keys(userInfo).length">
        <div class="app-menu__item"
             v-for="item in headerPanelMenu" :key="item.id">
          <headerMenuItem  :item="item"/>
        </div>
      </template>
      <template v-else>
        <div class="app-menu__item"
             v-for="item in headerPanelMenuLogged" :key="item.id">
          <headerMenuItem  :item="item"/>
        </div>
      </template>
    </template>
  </div>
</template>

<script>
import headerPanelMenu from '@/constants/headerMenu';
import headerMenuItem from '@/components/header/headerMenuItem.vue';
import headerPanelMenuLogged from '@/constants/headerMenuLogged';

export default {
  name: 'HeaderMenuTemplate',
  components: {
    headerMenuItem,
  },
  data() {
    return {
      headerPanelMenu,
      headerPanelMenuLogged,
    };
  },
  computed: {
    userInfo: {
      get() {
        return this.$store.getters.userInfo;
      },
      set(data) {
        this.$store.dispatch('setUserInfo', data);
      },
    },
  },
};
</script>

<style scoped lang="scss">
@import "../../assets/scss/variables";

.app-menu {
  display: flex;
  align-items: center;

  &__item {
    cursor: pointer;
    transition: opacity .15s ease-in;

    &:hover {
      opacity: .8;
    }
  }
}
</style>
