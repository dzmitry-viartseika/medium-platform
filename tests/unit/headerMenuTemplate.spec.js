import { shallowMount, createLocalVue, render } from '@vue/test-utils';
import headerMenuTemplate from '@/components/header/headerMenuTemplate.vue';

import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);
const store = new Vuex.Store({});

describe('headerMenuTemplate.vue', () => {
  test('snapshot', () => {
    const $store = {
      state: {
        userInfo: 1,
      },
      getters: {
        userInfo: () => ({}),
      },
    };
    const wrapper = shallowMount(headerMenuTemplate, {
      localVue,
      store,
      render,
      mocks: {
        $store,
      },
    });
    expect(wrapper.find('.app-menu')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
