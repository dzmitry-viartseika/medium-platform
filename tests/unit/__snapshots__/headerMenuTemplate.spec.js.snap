// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headerMenuTemplate.vue snapshot 1`] = `
<div class="app-menu">
  <!---->
</div>
`;
